android/app/src/main/java/org/uwaterloo/subletr/pages/home/HomePageView.kt

```diff
@@ -104,14 +104,33 @@ fun HomePageView(
 			CircularProgressIndicator()
 		}
 	} else if (uiState is HomePageUiState.Loaded) {
-		// TODO: Ensure that this is the best way to refresh on reload
-		LaunchedEffect(Unit) {
-			viewModel.priceRangeFilterStream.onNext(uiState.priceRange)
-		}
-
 		val listState = rememberLazyListState()
 		val isListView = remember { mutableStateOf(true) }
 
+		val lastItemIsShowing by remember {
+			derivedStateOf {
+				if (listState.layoutInfo.totalItemsCount == 0) {
+					false
+				}
+				else {
+					listState.layoutInfo.visibleItemsInfo.lastOrNull()?.index == listState.layoutInfo.totalItemsCount - 1
+				}
+			}
+		}
+
+		LaunchedEffect(key1 = lastItemIsShowing) {
```

Comment:
Just out of curiosity, what is key1 here?

Reply:
`key1` basically means that the side effect in the `LaunchedEffect` will run every time the `key1` is updated.

Not the optimal solution probably, but it should work.